packages/docs-gesture-handler/docs/gestures/use-fling-gesture.mdx

```diff
@@ -0,0 +1,189 @@
+---
+id: use-fling-gesture
+title: Fling gesture
+sidebar_label: Fling gesture
+sidebar_position: 6
+---
+
+import { vanishOnMobile, appearOnMobile, webContainer } from '@site/src/utils/getGestureStyles';
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+import FlingGestureBasic from '@site/static/examples/FlingGestureBasic';
+import FlingGestureBasicSrc from '!!raw-loader!@site/static/examples/FlingGestureBasicSrc';
+
+<div className={webContainer}>
+  <div className={vanishOnMobile} style={{ display: 'flex', justifyContent: 'center', maxWidth: 360 }}>
+    <video playsInline autoPlay muted loop style={{maxWidth: 360}}>
+      <source src={useBaseUrl("/video/fling.mp4")} type="video/mp4"/>
+    </video>
+  </div>
+  <InteractiveExample
+    component={<FlingGestureBasic/>}
+    src={FlingGestureBasicSrc}
+    disableMarginBottom={true}
+  />
+</div>
+
+import BaseEventData from './\_shared/base-gesture-event-data.md';
+import BaseGestureConfig from './\_shared/base-gesture-config.md';
+import BaseGestureCallbacks from './\_shared/base-gesture-callbacks.mdx';
+
+A discrete gesture that activates when the movement is sufficiently long and fast.
+Gesture gets [ACTIVE](/docs/fundamentals/states-events#active) when movement is sufficiently long and it does not take too much time.
+When gesture gets activated it will turn into [END](/docs/fundamentals/states-events#end) state when finger is released.
+The gesture will fail to recognize if the finger is lifted before being activated.
+
+  <div className={appearOnMobile} style={{ display: 'flex', justifyContent: 'center' }}>
+    <video playsInline autoPlay muted loop style={{maxWidth: 360}}>
+      <source src={useBaseUrl("/video/fling.mp4")} type="video/mp4"/>
+    </video>
+  </div>
+
+## Example
+
+```jsx
+import { StyleSheet } from 'react-native';
+import {
+  GestureDetector,
+  GestureHandlerRootView,
+  Directions,
+  useFlingGesture,
+} from 'react-native-gesture-handler';
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withTiming,
+} from 'react-native-reanimated';
+
+export default function App() {
```

Comment:
This is different from both of the above, but would require composition to work. I'm not sure if we want to include that here.